qlty-check/src/planner/config_files.rs

```diff
@@ -148,3 +175,614 @@ pub fn plugin_configs(planner: &Planner) -> Result<HashMap<String, Vec<PluginCon
+                destination: planner.staging_area.destination_directory.join(file_name),
+                mode: ConfigCopyMode::Symlink,
+            });
+        }
+
+        operations.push(ConfigOperation {
+            source: ConfigSource::File(exported_config_path.clone()),
+            destination: planner.workspace.root.join(file_name),
+            mode: ConfigCopyMode::Symlink,
+        });
+    }
+
+    Ok(operations)
+}
+
+fn qlty_config_operations(
+    planner: &Planner,
+    plugins: &[ActivePlugin],
+) -> Result<Vec<ConfigOperation>> {
+    let mut operations = Vec::new();
+    let library = planner.workspace.library().ok();
+
+    let config_file_names: Vec<_> = plugins
+        .iter()
+        .flat_map(|plugin| &plugin.plugin.config_files)
+        .map(|path| path.to_string_lossy().to_string())
+        .collect::<std::collections::HashSet<_>>()
+        .into_iter()
+        .collect();
+
+    for config_file_name in config_file_names {
+        let source_path = if let Some(library) = &library {
+            library.configs_dir().join(&config_file_name)
+        } else {
+            PathBuf::from(&config_file_name)
+        };
+
+        if planner.workspace.root != planner.staging_area.destination_directory {
+            operations.push(ConfigOperation {
+                source: ConfigSource::File(source_path.clone()),
+                destination: planner
+                    .staging_area
+                    .destination_directory
+                    .join(&config_file_name),
+                mode: ConfigCopyMode::Symlink,
+            });
+        }
+
+        operations.push(ConfigOperation {
+            source: ConfigSource::File(source_path.clone()),
+            destination: planner.workspace.root.join(&config_file_name),
+            mode: ConfigCopyMode::Symlink,
+        });
+    }
+
+    Ok(operations)
+}
+
+fn fetch_operations(planner: &Planner, plugins: &[ActivePlugin]) -> Vec<ConfigOperation> {
+    let mut operations = Vec::new();
+
+    for active_plugin in plugins {
+        for fetch in &active_plugin.plugin.fetch {
+            operations.push(ConfigOperation {
+                source: ConfigSource::Download(fetch.clone()),
+                destination: planner.workspace.root.join(&fetch.path),
+                mode: ConfigCopyMode::Copy,
+            });
+
+            if planner.workspace.root != planner.staging_area.destination_directory {
+                operations.push(ConfigOperation {
+                    source: ConfigSource::Download(fetch.clone()),
+                    destination: planner.staging_area.destination_directory.join(&fetch.path),
+                    mode: ConfigCopyMode::Copy,
+                });
+            }
+        }
+    }
+
+    operations
+}
+
+pub fn compute_tool_install_config_operations(
+    planner: &Planner,
+    invocations: &[crate::planner::InvocationPlan],
+) -> Result<Vec<ConfigOperation>> {
```

Comment:
I was expecting that all of the config operations would be exposed through the `compute_config_staging_operations` which is public.

However, this function is also public, which suggests something else in the codebase is calling it directly.

This seems to break the abstraction that we are providing one function (`compute_config_staging_operations`) which produces _all_ of the config operations.

Why is `compute_tool_install_config_operations` exposed seprately?